packages/gasket-plugin-dynamic-plugins/README.md

```diff
@@ -0,0 +1,77 @@
+# @gasket/plugin-dynamic-plugins
+
+Gasket plugin for dynamically adding plugins to Gasket
+
+## Installation
+
+```
+npm i @gasket/plugin-dynamic-plugins
+```
+
+Update your `gasket` file plugin configuration:
+
+```diff
+// gasket.js
+
++ import pluginDynamicPlugins from '@gasket/plugin-dynamic-plugins';
+
+ export default makeGasket({
+  plugins: [
++   pluginDynamicPlugins
+  ]
+});
+```
+
+## Usage
+
+This plugin gives you the option to dynamically add plugins to Gasket after the core Gasket instance has been created.
+
+This is can be useful when you want to add plugins to Gasket only in specific environments.
+
+For example, if you have plugins for docs creation (`@gasket/plugin-docs`, `@gasket/plugin-docusarus`) that are only needed for development purposes and do not need to be included in your production code, you can dynamically load them into Gasket only when doing development locally through this plugin. You could then add plugins designated only for development to you `devDependencies` in your `package.json` file.
+
+If you want to wait until dynamic plugins have been loaded into the Gasket instance before running app code, you can do so by checking if the `isReady` property on the Gasket instance has been resolved.
+
+```js
+import gasket from './gasket.js';
+gasket.isReady.then(() => {
+  gasket.actions.startServer();
+});
+```
+
+## Configuration
+
+### Environment Variables
+
+#### `GASKET_DYNAMIC`
+
+The `GASKET_DYNAMIC` environment variable is used to control the loading of dynamic plugins in the Gasket framework. When this variable is set, the framework will attempt to load and register dynamic plugins specified in the configuration.
+
+To enable dynamic plugin loading, set the `GASKET_DYNAMIC` environment variable to `1`.
+
+```json
+"docs": "GASKET_DYNAMIC=1 node gasket.js docs"
+```
```

Comment:
I'm still hesitant with introducing a new env var. Given the example here, it seems like the existing `GASKET_ENV` would suffice. We've thus far used it for deploy environments, but we could easily utilize the [sub environment](https://github.com/godaddy/gasket/blob/main/docs/configuration.md#environments) for things like running `docs` or `analyze` locally.

```
GASKET_ENV=local.docs node gasket.js docs
```
with

```js
makeGasket({
  environments: {
    local: { ... },
    local.docs: {
      dynamicPlugins: [
        '@gasket/plugin-docs',
        '@gasket/plugin-docusaurus'
      ]
    }
  }
})
```

Reply:
I wonder if we could bring back [command-based config](https://github.com/godaddy/gasket/blob/main/docs/configuration.md#environments) from the old cli approach... Not blocking for this PR. Just a thought of something we should investigate eventually.